### ISSUE TITLE
Add a remove or delete setting API

### ISSUE THREAD

**Issue by kaelad02** [foundryvtt/foundryvtt#8479]
### User Experience

TL;DR: Can we add an API to `ClientSettings` to remove settings?

For one of my modules, I found a better way to handle some of my settings. Instead of separate settings shown on the normal settings app, I wanted some custom handling so I created a menu using the `ClientSettings#registerMenu` API. Then I created a `FormApplication` and all is well. Turns out there's some further changes I could do by moving some of the settings into one, as an object, since they're changed in concert and I'd really like to have just one `onChange` function called instead of two.

I can create the new setting but I'm having trouble migrating the settings, mainly removing the old settings. I want to remove the old settings that way I don't try and migrate the settings again and possibly override the new settings. Looking at the API docs and in the console, `ClientSettings` doesn't have a function to remove settings. I found a way with some help on the Discord, but it would be helpful if there was an official API call to do this. If we're doing this, we should handle both regular settings as well as menus (i.e. both `register` and `registerMenu`). Not sure it would need two API calls to remove, just one would be fine that worked for both IMHO.

**Comment by aaclayton** [comment]
There is already an API you can use for this, although it's slightly more obscure than in the case of other document types. What you need to do depends on the scope of the setting you want to delete.

## Client Settings
Client settings use the standard web storage API to store data locally in the user's browser. Settings which use client storage, like whether or not to display the FPS meter, can be deleted by removing the item from the browser's local storage:
```js
localStorage.deleteItem("core.fpsMeter")
```

## World Settings
World scope settings are stored on the server-side in a Document database. You can retrieve the actual `Setting` document by the setting key as follows:
```js
const worldSettings = game.settings.storage.get("world");
const setting = worldSettings.getSetting("dnd5e.systemMigrationVersion");
```
That `setting` is a `Document` that can be modified or deleted using the standard Document APIs:
```js
setting.delete();
```

**Comment by kaelad02** [comment]
Mine were client settings and that code is what I was using to remove it. It took someone some digging into the internals of how settings are stored to figure out how to delete them. I just thought it would be better to have something more "official" in the `ClientSettings` class since that's where you register, get, and set settings.
